### DISCUSSION TITLE
Send events from a spawned actor to the parent(another spawned actor)

### DISCUSSION THREAD

**Discussion by Kashif7** [statelyai/xstate#2257] [Q&A] [answered]
sandbox link: https://codesandbox.io/s/trusting-lehmann-m59pf?file=/src/machines.js

I have a machine(grandparent) that is spawning an actor(parent), and then, in turn, that actor spawns another actor(child). I need to send an event from the child actor to the parent actor. But when I use "sendParent" to achieve this, the event always goes to the grandparent, even when I assign a name to the parent actor and pass it as the "to" param. I tried using "send" instead of "sendParent" but that results in an error. What am I doing wrong? :/

**Comment by davidkpiano** [comment, marked as answer]
Sorry, we should definitely make these issues clearer! When you spawn, it should always be "deferred" (in a function):

```diff
        entry: assign({
-         child: spawn(createChild())
+         child: () => spawn(createChild())
        }),
```

That will solve your problem.

**Reply by Kashif7** [reply]
Yes, that fixed the issue 😄 Thanks! I just checked and it is mentioned in docs, but I have missed it. My bad 😅
